### PR TITLE
add parallel execution overview to lib.rs as doc

### PR DESCRIPTION
**Pull request by gelash**
Adding algorithm overview as documentation to the lib.rs

## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/aptos-labs/aptos-core/tree/main/developer-docs-site, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for the release.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

**Comment by sitalkedia** [comment]
Thanks for adding very crisp documentation for parallel execution @gelash. Left minor nits.

**Review by zekun000 — APPROVED** [review]

File: aptos-move/parallel-executor/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,138 @@
...
+all transactions in the block. A transaction tx not in E is either currently
+being executed or (its last incarnation) has completed.
+Each thread repeats the following (loop in 'executor.rs'):
+
+- Check done: if V and E are empty and no other thread is performing a task,
+then return.
+- Find next task: Perform the task with the smallest transaction index tx in V
+and E:
+  1. Execution task: Execute the next incarnation of tx. If a value marked as
+     ESTIMATE is read, abort execution and add tx back to E. Otherwise:
```
[the remark this answers is not in the document]
Comment (gelash) [reply aptos-move/parallel-executor/src/lib.rs:76]: added clarification to the ordered set.

File: aptos-move/parallel-executor/src/lib.rs [hunk 2]
```diff
@@ -1,6 +1,138 @@
...
+     (a) If there is a write to a memory location to which the previous finished
+         incarnation of tx has not written, create validation tasks for all
+         transactions >= tx that are not currently in E or being executed and
+         add them to V.
+     (b) Otherwise, create a validation task only for tx and add it to V.
```
[the remark this answers is not in the document]
Comment (gelash) [reply aptos-move/parallel-executor/src/lib.rs:81]: the ordered set semantics disregards duplicates - hopefully the clarification I added helps here too.

**Comment by gelash** [comment]
reopen

**Pull request closed without merging**